server/src/main/java/org/opensearch/index/compositeindex/datacube/DimensionFactory.java

```diff
@@ -69,8 +72,10 @@ public static Dimension parseAndCreateDimension(
                 return parseAndCreateDateDimension(name, dimensionMap, c);
             case NUMERIC:
                 return new NumericDimension(name);
-            case KEYWORD:
-                return new KeywordDimension(name);
+            case ORDINAL:
+                return new OrdinalDimension(name);
+            case IP:
```

Comment:
This switch code block is same as in the other `parseAndCreateDimension `. Let's avoid duplication.